lms/djangoapps/discussion_api/api.py

```diff
@@ -901,7 +913,9 @@ def update_thread(request, thread_id, update_data):
         The updated thread; see discussion_api.views.ThreadViewSet for more
         detail.
     """
-    cc_thread, context = _get_thread_and_context(request, thread_id)
+    cc_thread, context = _get_thread_and_context(
+        request, thread_id, retrieve_kwargs={"user_id": unicode(request.user.id)}
```

Comment:
@wajeeha-khalid: We were discussing offline if this should also always return True for read as a reasonable shortcut, without incurring additional processing.  Let me know if you have any concerns with that.